### ISSUE TITLE
Fix S2583 FP: Number constraint from remainder calculation is not correct.

### ISSUE THREAD

**Issue by konrad-gora** [SonarSource/sonar-dotnet#8285]
### Description

S2583:Conditionally executed code should be reachable

It shows a warning for if with condition that compares int with 0

### Repro steps

Example of code that shows a warning:
```
public static BigInteger Calculate(int number)
{
    if (number < 0) return number % 2 == 0 ? 2 : 3;
    return 12;
}
```

### Expected behavior

Don't show a warning in this case.

### Actual behavior

There is a warning :)

### Known workarounds

Just suppress.

### Related information

* C#/VB.NET Plugins version - 7.3.0.77872
* Visual Studio version - 17.7.6
* MSBuild / dotnet version - net7.0
* SonarScanner for .NET version (if used) - not used
* Operating System - Windows 10

**Comment by mary-georgiou-sonarsource** [comment]
Hello @konrad-gora,
Thanks a lot for reporting this, I confirm it's an FP.
I'll add the issue to our backlog.

**Comment by mary-georgiou-sonarsource** [comment]
This FP is raising because the remainder constraint is `[-1]` instead of `[-1, 0]`.
